src/sentry/static/sentry/app/components/resultGrid.tsx

```diff
@@ -1,59 +1,62 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import createReactClass from 'create-react-class';
-import $ from 'jquery';
 import {browserHistory} from 'react-router';
+import {Location} from 'history';
 
 import withApi from 'app/utils/withApi';
 import DropdownLink from 'app/components/dropdownLink';
 import MenuItem from 'app/components/menuItem';
 import Pagination from 'app/components/pagination';
 import {IconSearch} from 'app/icons';
+import {Client, RequestOptions} from 'app/api';
 
-class Filter extends React.Component {
-  static propTypes = {
-    name: PropTypes.string.isRequired,
-    queryKey: PropTypes.string.isRequired,
-    options: PropTypes.array.isRequired,
-    path: PropTypes.string.isRequired,
-    value: PropTypes.any,
-  };
+type Option = [string, string];
 
-  getCurrentLabel = () => {
+type FilterProps = {
+  name: string;
+  queryKey: string;
+  options: Option[];
+  path: string;
+  value: string;
+  location: Location;
+};
+
+class Filter extends React.Component<FilterProps> {
+  getCurrentLabel() {
     const selected = this.props.options.filter(
-      item => item[0] === (this.props.value || '')
+      item => item[0] === (this.props.value ?? '')
     )[0];
     if (selected) {
       return this.props.name + ': ' + selected[1];
     }
     return this.props.name + ': ' + 'Any';
-  };
+  }
 
-  getDefaultItem = () => {
-    const query = $.extend({}, this.props.location.query, {
-      cursor: '',
-    });
+  getDefaultItem() {
+    const query = Object.assign({}, this.props.location.query, {cursor: ''});
     delete query[this.props.queryKey];
 
     return (
       <MenuItem
         key=""
-        isActive={this.props.value === '' || !this.props.value}
+        isActive={this.props.value === '' ?? !this.props.value}
```

Comment:
please, is it really necessary to check for `this.props.value === '' ` ? I would write something like `!this.props.value.trim()`

Reply:
I don't think these are functionally equivalent. The `??` operator only uses the right hand side if the left is null/undefined which can never happen in a boolean operation.

Reply:
Oh yeah, this is a mistake 👍